packages/ckeditor5-link/src/autolink.ts

```diff
@@ -104,6 +107,54 @@ export default class AutoLink extends Plugin {
 	public afterInit(): void {
 		this._enableEnterHandling();
 		this._enableShiftEnterHandling();
+		this._enablePasteLinking();
+	}
+
+	/**
+	 * Enables autolinking on pasting a URL.
+	 */
+	private _enablePasteLinking(): void {
+		const editor = this.editor;
+		const model = editor.model;
+		const selection = model.document.selection;
+		const clipboardPipeline = editor.plugins.get( 'ClipboardPipeline' );
+		const linkCommand = editor.commands.get( 'link' )!;
+
+		const updateLink = ( newLink: string ): boolean => {
+			const selStart = selection.getFirstPosition()!;
+			const selEnd = selection.getLastPosition()!;
+
+			const linkRange = findAttributeRange( selStart, 'linkHref', selection.getAttribute( 'linkHref' ), model );
+			// why is containsPosition false when the point is equal to the start or end of the range?
```

Comment:
> Format for comments

Sorry, I wasn't too concerned about comment formatting when this was intended as a question to be resolved not something to keep in the final code.

> Well, that's a decision what "contains" means

This is a crucial question, because a caret at the position you describe could be either inside or outside depending on your point of view. There's a technical difference, but no visual difference to the user. If the intent with CK is for start and end points to never be inside a "contains" check that's fine - I just wanted to check.

> Don't you want to check here whether the selection range intersects with the link range?

An intersection check is entirely what this code is doing. I don't check start because the range comes from `findAttributeRange` at `selStart`, which covers that side of the intersection. I'll take a closer look at `containsRange`, I didn't spot that when I was developing this so yes help with intersection checks will greatly simplify this code.

> selection may have multiple ranges

Ah, I wondered how you did that. In TinyMCE table selection is a side concept rather than add complexity to the core selection API. I'll add a check.